homeassistant/components/comelit/humidifier.py

```diff
@@ -0,0 +1,215 @@
+"""Support for humidifiers."""
+from __future__ import annotations
+
+from enum import StrEnum
+from typing import Any
+
+from aiocomelit import ComelitSerialBridgeObject
+from aiocomelit.const import CLIMATE
+
+from homeassistant.components.humidifier import (
+    MODE_AUTO,
+    MODE_NORMAL,
+    HumidifierAction,
+    HumidifierDeviceClass,
+    HumidifierEntity,
+    HumidifierEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ComelitSerialBridge
+
+
+class HumidifierComelitMode(StrEnum):
+    """Serial Bridge humidifier modes."""
+
+    AUTO = "A"
+    OFF = "O"
+    LOWER = "L"
+    UPPER = "U"
+
+
+class HumidifierComelitAction(StrEnum):
+    """Serial Bridge humidifier actions."""
+
+    OFF = "off"
+    ON = "on"
+    MANUAL = "man"
+    SET = "set"
+    AUTO = "auto"
+    LOWER = "lower"
+    UPPER = "upper"
+
+
+MODE_TO_ACTION: dict[str, HumidifierComelitAction] = {
+    MODE_AUTO: HumidifierComelitAction.AUTO,
+    MODE_NORMAL: HumidifierComelitAction.MANUAL,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit humidifiers."""
+
+    coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[ComelitHumidifierEntity] = []
+    for device_class in (
+        HumidifierDeviceClass.HUMIDIFIER,
+        HumidifierDeviceClass.DEHUMIDIFIER,
+    ):
+        entities.extend(
+            ComelitHumidifierEntity(
+                coordinator, device, config_entry.entry_id, device_class.value
+            )
+            for device in coordinator.data[CLIMATE].values()
+        )
+
+    async_add_entities(entities)
+
+
+class ComelitHumidifierEntity(CoordinatorEntity[ComelitSerialBridge], HumidifierEntity):
+    """Humidifier device."""
+
+    _attr_supported_features = HumidifierEntityFeature.MODES
+    _attr_available_modes = [MODE_NORMAL, MODE_AUTO]
+    _attr_min_humidity = 10
+    _attr_max_humidity = 90
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: ComelitSerialBridge,
+        device: ComelitSerialBridgeObject,
+        config_entry_entry_id: str,
+        device_class: str,
+    ) -> None:
+        """Init light entity."""
+        self._api = coordinator.api
+        self._device = device
+        super().__init__(coordinator)
+        # Use config_entry.entry_id as base for unique_id
+        # because no serial number or mac is available
+        self._attr_unique_id = f"{config_entry_entry_id}-{device.index}-{device_class}"
+        self._attr_name = f"{device_class.capitalize()}"
```

Comment:
This looks strange. I assume humidifier support entity name translations. This should set has entity name, the  the device class translation should be automatic?

Reply:
Right, so separate translation keys then.

Reply:
Fixed